### PR TITLE
Ask VA 1245 - Fix issues on Health facility and Residence pages

### PR DESCRIPTION
**Pull request by hemeshvpatel**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to Summary and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

## Summary

- Fix issues on health facility and residence pages per [figma](https://www.figma.com/design/aQ6JsjD4pvMxSVPAZHllMX/AVA-Page-Library?node-id=3478-53384&t=nlKB5dox8ayvWVNf-1) and accessibility testing 
- Updated the logic for Residence page for AT study
- The Ask VA team owns this page
- This is part of our POC

## Related issue(s)

- Link to ticket here: https://github.com/department-of-veterans-affairs/ask-va/issues/1245
- Work in this PR covers this ticket as well: https://github.com/department-of-veterans-affairs/ask-va/issues/1180

## Testing done

- Manual testing

## Screenshots

image

## What areas of the site does it impact?

- This only impacts the Ask VA form

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

**Review by joehall-tw — DISMISSED** [review]
LGTM.

File: src/applications/ask-va/components/search/SearchControls.jsx [hunk 1]
```diff
@@ -83,10 +83,11 @@ const SearchControls = props => {
               type="button"
               className="use-my-location-link"
             >
-              <i
+              <va-icon
+                size={4}
+                icon="null"
```
Comment (micahchiang) [inline src/applications/ask-va/components/search/SearchControls.jsx:88]: This isn't a valid value for va-icon. Please check the [va-icon guidance](https://design.va.gov/storybook/?path=/docs/uswds-va-icon--docs) to determine which icon you need.

File: src/applications/ask-va/components/search/SearchControls.jsx [hunk 2]
```diff
@@ -66,8 +66,8 @@ const SearchControls = props => {
             htmlFor="street-city-state-zip"
             id="street-city-state-zip-label"
           >
-            City, state or postal code{' '}
-            <span className="form-required-span">(*Required)</span>
+            City or postal code{' '}
+            {/* <span className="form-required-span">(*Required)</span> */}
```
Comment (micahchiang) [inline src/applications/ask-va/components/search/SearchControls.jsx:70]: can this be removed?

File: src/applications/ask-va/components/MedicalFacilitySearch.jsx [hunk 3]
```diff
@@ -5,9 +5,125 @@ import { convertToLatLng } from '../utils/mapbox';
...
+//       'https://dev-api.va.gov/ask_va_api/v0/health_facilities?lat=40.553874&long=-105.087951&page=1&per_page=10&state=CO&type=health',
+//     first:
+//       'https://dev-api.va.gov/ask_va_api/v0/health_facilities?lat=40.553874&long=-105.087951&per_page=10&state=CO&type=health',
+//     prev: null,
+//     next:
+//       'https://dev-api.va.gov/ask_va_api/v0/health_facilities?lat=40.553874&long=-105.087951&page=2&per_page=10&state=CO&type=health',
+//     last:
+//       'https://dev-api.va.gov/ask_va_api/v0/health_facilities?lat=40.553874&long=-105.087951&page=139&per_page=10&state=CO&type=health',
+//   },
+// };
```
Comment (micahchiang) [inline src/applications/ask-va/components/MedicalFacilitySearch.jsx:124]: Is this meant to serve as a reference for how data is structured? If so, consider keeping in `/fixtures/data` folder similar to [this example](https://github.com/department-of-veterans-affairs/vets-website/blob/5340e2608d43879ca1c897dcdd59f36a7776df03/src/applications/appeals/996/tests/fixtures/data/maximal-test-v1.json#L2)

**Review by micahchiang — CHANGES_REQUESTED** [review]

File: src/applications/ask-va/components/search/SearchControls.jsx (hunk 1 above)
Comment (hemeshvpatel) [reply src/applications/ask-va/components/search/SearchControls.jsx:88]: Updated to near_me icon per guidance

File: src/applications/ask-va/components/search/SearchControls.jsx (hunk 2 above)
Comment (hemeshvpatel) [reply src/applications/ask-va/components/search/SearchControls.jsx:70]: removed

File: src/applications/ask-va/components/MedicalFacilitySearch.jsx (hunk 3 above)
Comment (hemeshvpatel) [reply src/applications/ask-va/components/MedicalFacilitySearch.jsx:124]: Was for the data structure temporarily, removed

**Review by micahchiang — APPROVED** [review]

**Review by joehall-tw — APPROVED** [review]
Read it. Looks good. Thanks, Hemesh!

**Pull request merged**